File: src/Tests/SonarLint.UnitTest/TestCases/FlagsEnumWithoutInitializer.vb
Namespace Tests.Diagnostics

    <System.Flags()>
    Enum FruitType    ' Noncompliant
        Banana
        Orange = 5
        Strawberry
    End Enum

    Enum FruitType2    ' Compliant
        Banana
        Orange
        Strawberry
    End Enum

    <System.Flags()>
    Enum FruitType3
        Banana = 1
        Orange = 3
        Strawberry = 4
    End Enum
End Namespace

File: src/Tests/SonarLint.UnitTest/TestCases/FlagsEnumWithoutInitializer.cs
using System;
namespace Tests.Diagnostics
{
    [System.Flags]
    enum FruitType    // Noncompliant {{Initialize all the members of this enumeration.}}
//       ^^^^^^^^^
    {
        Banana,
        Orange = 5,
        Strawberry
    }
    enum FruitType2    // Compliant
    {
        Banana,
        Orange,
        Strawberry
    }

    [Flags]
    enum FruitType3    // Compliant
    {
        Banana=1,
        Orange =4,
        Strawberry =5
    }
}
